unit_test/platform_test.py

```diff
@@ -1,14 +1,29 @@
 import sys
 import os
+import subprocess
 
 import pytest
 
 from cibuildwheel.__main__ import main
+from cibuildwheel import windows, linux, macos
 
 
-def test_unknown_platform_non_ci(monkeypatch, capsys):
+def not_call_mock(*args, **kwargs):
+    raise RuntimeError("This should never be called")
+
+
+def apply_mock_protection(monkeypatch):
+    monkeypatch.setattr(subprocess, "Popen", not_call_mock)
+    monkeypatch.setattr(windows, "urlopen", not_call_mock)
```

Comment:
What's this urlopen mock for - could it be removed?

Reply:
on windows urlopen is used to do download nuget on disc
https://github.com/joerick/cibuildwheel/blob/054be2eff2293dfa9f3c380a9d20b9c91786b2e1/cibuildwheel/windows.py#L70

https://github.com/joerick/cibuildwheel/blob/054be2eff2293dfa9f3c380a9d20b9c91786b2e1/cibuildwheel/windows.py#L95-L99

and I decide to mock any function that can introduce change on machine.
I can remove it if you wish.